### PR TITLE
Pull request for Eventgrid

### PR DESCRIPTION
**Pull request by adityapawan-b**
Event Grid subscription created using function as endpoint. Created new version of module V1.1 and added to Modules

### CODE REVIEW

File: services/frontdoor/firewallPolicy/base/v1/main.tf [hunk 1]
```diff
@@ -13,10 +13,30 @@ resource "azurerm_frontdoor_firewall_policy" "waf" {
...
+		  version = managed_rule.value.version
+      match_condtion {
+        
+      }
 		}
 	}
-  
+
+  dynamic "custom_rule" {
+    for_each = var.custom_rules_settings
```
Comment (Steve-Persistent) [inline services/frontdoor/firewallPolicy/base/v1/main.tf:24]: This custom_ruie doesn't have an ending brace.

Comment (adityapawan-b) [reply services/frontdoor/firewallPolicy/base/v1/main.tf:24]: Hi Steve,

That custom ruleset is still not completed so it's commented out and will be coming later after full testing on that block of code

File: services/eventgrid/subscription/fn/v1.1/variables.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  type = object({
+    
+    function_name        = string
+    
+  })
+}
+variable "observability_settings" {
+  type = object({
+    instrumentation_key       = string
+    workspace_id              = string
```
Comment (Steve-Persistent) [inline services/eventgrid/subscription/fn/v1.1/variables.tf:31]:
```suggestion
    We have added a few more settings to observability_settings, including storage account to also send events to.  We need to add this eventually to this module.  It's a bit out of date right now.
```

File: services/frontdoor/firewallPolicy/base/v1/variables.tf [hunk 3]
```diff
@@ -32,4 +32,23 @@ variable "managed_rules_settings" {
         type    = string
         version = string
     }))
+}
+
+variable "custom_rules_settings" {
```
Comment (Steve-Persistent) [inline services/frontdoor/firewallPolicy/base/v1/variables.tf:37]: This looks painful, and should be a map.  The module isn't using it, so can we comment this out and go back and clean up.

File: services/frontdoor/firewallPolicy/base/v1/main.tf (hunk 1 above)
Comment (adityapawan-b) [reply services/frontdoor/firewallPolicy/base/v1/main.tf:24]: corrected the syntax and updated the pull

**Comment by Steve-Persistent** [comment]
Adding event grid subscriptions.

**Pull request merged**